dagshub/data_engine/model/datasource.py

```diff
@@ -343,6 +345,31 @@ def send_datapoints_to_annotation(self, datapoints: Union[List[Datapoint], List[
             webbrowser.open_new_tab(link)
         return link
 
+    def wait_until_ready(self, max_wait_time=None):
+        """
+       Blocks until the datasource preprocessing is complete
+
+       Args:
+           max_wait_time (int): Maximum time to wait in seconds
+       """
+        start = time.time()
+        logger.info("Waiting for datasource preprocessing to complete")
```

Comment:
logger.info() is not seen by default.
Check out this: https://rich.readthedocs.io/en/stable/reference/status.html if you want something nicer (don't forget to use the `rich_console` from the common imports so it's not displayed in quiet mode)